File: backend/src/main/java/com/java/NBE4_5_1_7/domain/news/dto/responseDto/JobsDetailDto.java
package com.java.NBE4_5_1_7.domain.news.dto.responseDto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;
import lombok.Getter;

import java.util.List;

@Data
@Getter
@JsonIgnoreProperties(ignoreUnknown = true)
public class JobsDetailDto {
    private String recrutPblntSn;
    private String pblntInstCd;
    private String pbadmsStdInstCd;
    private String instNm;
    private String ncsCdLst;
    private String ncsCdNmLst;
    private String hireTypeLst;
    private String hireTypeNmLst;
    private String workRgnLst;
    private String workRgnNmLst;
    private String recrutSe;
    private String recrutSeNm;
    private String prefCondCn;
    private int recrutNope;
    private String pbancBgngYmd;
    private String pbancEndYmd;
    private String recrutPbancTtl;
    private String srcUrl;
    private String replmprYn;
    private String aplyQlfcCn;
    private String disqlfcRsn;
    private String scrnprcdrMthdExpln;
    private String prefCn;
    private String acbgCondLst;
    private String acbgCondNmLst;
    private String nonatchRsn;
    private String ongoingYn;
    private int decimalDay;
    private List<Files> files;
    private List<Steps> steps;

    @Data
    @JsonIgnoreProperties(ignoreUnknown = true)  // 여기에도 추가
    public static class Files {
        private int recrutAtchFileNo;  // 필드 추가
        private int sortNo;
        private String atchFileNm;
        private String atchFileType;
        private String url;
    }

    @Data
    @JsonIgnoreProperties(ignoreUnknown = true)  // 여기에도 추가
    public static class Steps {
        private int recrutStepSn;
        private int recrutPblntSn;
        private String recrutPbancTtl;
        private String recrutNope;
        private String aplyNope;
        private String cmpttRt;
        private String rsnOcrnYmd;
        private int sortNo;
        private int minStepSn;
        private int maxStepSn;
    }
}

File: backend/src/main/java/com/java/NBE4_5_3_7/domain/news/dto/responseDto/JobsDetailDto.kt
package com.java.NBE4_5_3_7.domain.news.dto.responseDto

import com.fasterxml.jackson.annotation.JsonIgnoreProperties

@JsonIgnoreProperties(ignoreUnknown = true)
class JobsDetailDto {
    var recrutPblntSn: String? = null
    var pblntInstCd: String? = null
    var pbadmsStdInstCd: String? = null
    var instNm: String? = null
    var ncsCdLst: String? = null
    var ncsCdNmLst: String? = null
    var hireTypeLst: String? = null
    var hireTypeNmLst: String? = null
    var workRgnLst: String? = null
    var workRgnNmLst: String? = null
    var recrutSe: String? = null
    var recrutSeNm: String? = null
    var prefCondCn: String? = null
    var recrutNope: Int = 0
    var pbancBgngYmd: String? = null
    var pbancEndYmd: String? = null
    var recrutPbancTtl: String? = null
    var srcUrl: String? = null
    var replmprYn: String? = null
    var aplyQlfcCn: String? = null
    var disqlfcRsn: String? = null
    var scrnprcdrMthdExpln: String? = null
    var prefCn: String? = null
    var acbgCondLst: String? = null
    var acbgCondNmLst: String? = null
    var nonatchRsn: String? = null
    var ongoingYn: String? = null
    var decimalDay: Int = 0
    var files: List<Files>? = null
    var steps: List<Steps>? = null

    @JsonIgnoreProperties(ignoreUnknown = true)
    class Files {
        var recrutAtchFileNo: Int = 0 // 필드 추가
        var sortNo: Int = 0
        var atchFileNm: String? = null
        var atchFileType: String? = null
        var url: String? = null
    }

    @JsonIgnoreProperties(ignoreUnknown = true)
    class Steps {
        var recrutStepSn: Int = 0
        var recrutPblntSn: Int = 0
        var recrutPbancTtl: String? = null
        var recrutNope: String? = null
        var aplyNope: String? = null
        var cmpttRt: String? = null
        var rsnOcrnYmd: String? = null
        var sortNo: Int = 0
        var minStepSn: Int = 0
        var maxStepSn: Int = 0
    }

}

